### PR TITLE
[Timers] Follow timers from timelines

### PR DESCRIPTION
**Pull request by VWoeltjen**
Work-in-progress for #1688; still to-do:

- [x] Add time-of-interest controller
- [x] Follow time conductor from TOI controller
- [x] Synchronize zoom to time conductor bounds
- [x] Adjust time conductor times to SET using timer (for timelines)
- [x] Display TOI line
- [x] Add a `tick` event to the Time API (or find an alternate solution)
- [x] Don't follow timers while scrolling (try "gravity well" approach - follow when in bounds)
- [x] Reduce `$watch` count (these look unnecessary)
- [x] Handle `$destroy`
- [x] Reasonable bounds-following (don't clobber user changes; check zoom level?)
- [x] Remove tick/TOI following (isn't bounds-following enough?)
- [x] Proper CSS
- [x] ~~Degrade gracefully when time system is not UTC~~ (academic concern, ignoring for now)
- [x] Documentation (esp for new event, if added)
- [x] Tests
- [x] One more round of smoketesting
- [x] Remove `tick`
- [x] Get played state following working
- [x] Get paused state following working
- [x] Get stopped state to be ignored
- [x] Merge in frontend changes
- [x] Investigate [line misplacement during zoom](https://github.com/nasa/openmct/issues/1688#issuecomment-330373625)
- [x] Check behavior with big timer values
- [x] Resolve [build failure](https://circleci.com/gh/nasa/openmct/4181)
- [x] Smoketest
- [x] Build

### CODE REVIEW

**Comment by VWoeltjen** [comment]
@larkin - this is in a reasonable place to begin review. Still needs tests and code style cleanup, and should probably be squash-merged given the commit history. Added a button to follow time conductor bounds, automatically enabled/disabled based on zoom changes; this looked like the simplest solution short term.

There's a "degrade gracefully when time system is not UTC" in there that I'm tempted to move to a low-priority followup, since we don't have any non-UTC deployments using timelines to support.

File: src/api/time/TimeAPI.js [hunk 1]
```diff
@@ -329,6 +329,7 @@ define(['EventEmitter'], function (EventEmitter) {
 
         this.boundsVal = newBounds;
         this.emit('bounds', this.boundsVal, true);
+        this.emit('tick', timestamp);
```
Comment (larkin) [inline src/api/time/TimeAPI.js]: Hrm, don't want to add an extra event to the time API.  Note that the `bounds` event is fired with a `tick` argument that is `true` whenever it is coming from a clock.  You can use that to only react to ticks.

File: API.md [hunk 2]
```diff
@@ -755,6 +755,8 @@ The events emitted by the Time API are:
     If `tick` is false,then the bounds change was not due to an automatic tick, 
     and a query for historical data may be necessary, depending on your data 
     caching strategy, and how significantly the start bound has changed.
+* `tick`: emitted whenever an active clock ticks. The callback will be invoked
```
Comment (larkin) [inline API.md]: Don't need a new event for this when `bounds` carries a second argument to determine whether or not it was a clock `tick`.

File: platform/features/clock/src/actions/FollowTimerAction.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    [],
+    function () {
+
+        /**
+         * Sets the Time Conductor to follow a specific timer.
```
Comment (larkin) [inline platform/features/clock/src/actions/FollowTimerAction.js]: I would expect a timer would follow the time conductor, not the other way around.

File: API.md (hunk 2 above)
Comment (larkin) [inline API.md]: Don't need a new event for this when `bounds` carries a second argument to determine whether or not it was a clock `tick`.

File: src/api/time/TimeAPI.js (hunk 1 above)
Comment (larkin) [inline src/api/time/TimeAPI.js]: Hrm, don't want to add an extra event to the time API.  Note that the `bounds` event is fired with a `tick` argument that is `true` whenever it is coming from a clock.  You can use that to only react to ticks.

File: platform/features/clock/src/actions/FollowTimerAction.js (hunk 3 above)
Comment (VWoeltjen) [reply platform/features/clock/src/actions/FollowTimerAction.js]: It's kind of neither-nor and I need to rewrite this JSDoc. Time conductor doesn't really care about timer or vice versa

**Comment by larkin** [comment]
@VWoeltjen Rebase & squash on master to clean up commit history would be good.

**Comment by VWoeltjen** [comment]
@charlesh88 - this is ready for your attention! Time-of-interest div is currently [here](https://github.com/nasa/openmct/blob/7ab0693cc983f8a04ac8ee9002f4d776b06a869a/platform/features/timeline/res/templates/timeline.html#L142-L143) and should be reasonably easy to reposition and style. Needs to stay in the scope of the `toiController`

**Comment by charlesh88** [comment]
@VWoeltjen I am on this

**Comment by charlesh88** [comment]
@VWoeltjen I've pushed my changes to branch timeline-follow-1688-styling.

**Comment by VWoeltjen** [comment]
### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

**Comment by larkin** [comment]
# Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y

**Pull request merged**